### ISSUE TITLE
ImportError

### ISSUE THREAD

**Issue by mikecheng2626** [nischi/MMM-Face-Reco-DNN#141]
magicmirror@pi:~/MagicMirror/modules/MMM-Face-Reco-DNN $ npm run encode

> mmm-face-reco-dnn@2.0.1 encode
> python tools/encode.py -i dataset -e model/encodings.pickle

Traceback (most recent call last):
  File "/home/magicmirror/MagicMirror/modules/MMM-Face-Reco-DNN/tools/encode.py", line 4, in <module>
    import cv2
  File "/home/magicmirror/.local/lib/python3.9/site-packages/cv2/__init__.py", line 181, in <module>
    bootstrap()
  File "/home/magicmirror/.local/lib/python3.9/site-packages/cv2/__init__.py", line 153, in bootstrap
    native_module = importlib.import_module("cv2")
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: /home/magicmirror/.local/lib/python3.9/site-packages/cv2/cv2.abi3.so: undefined symbol: __atomic_store_8

I did not see a folder named "dataset" I created it on my own.

**Comment by nischi** [comment]
Hi @mikecheng2626 
Yes the folder dataset need to be created with some photos. https://github.com/nischi/MMM-Face-Reco-DNN#face-dataset
But maybe i need to create a better error message

**Comment by mirrormonark** [comment]
> magicmirror@pi:~/MagicMirror/modules/MMM-Face-Reco-DNN $ npm run encode
> 
> > mmm-face-reco-dnn@2.0.1 encode
> > python tools/encode.py -i dataset -e model/encodings.pickle
> 
> Traceback (most recent call last): File "/home/magicmirror/MagicMirror/modules/MMM-Face-Reco-DNN/tools/encode.py", line 4, in import cv2 File "/home/magicmirror/.local/lib/python3.9/site-packages/cv2/**init**.py", line 181, in bootstrap() File "/home/magicmirror/.local/lib/python3.9/site-packages/cv2/**init**.py", line 153, in bootstrap native_module = importlib.import_module("cv2") File "/usr/lib/python3.9/importlib/**init**.py", line 127, in import_module return _bootstrap._gcd_import(name[level:], package, level) ImportError: /home/magicmirror/.local/lib/python3.9/site-packages/cv2/cv2.abi3.so: undefined symbol: __atomic_store_8
> 
> I did not see a folder named "dataset" I created it on my own.

I have the same problem now. did you manage to solve it? @mikecheng2626

**Comment by TroyesDev** [comment]
Hi guys, same problem here, created the folder as told and everything else as per the instructions and same error

**Comment by nischi** [comment]
so you created a folder with name dataset with some subfolder with your names and inside of this named folders some pictures to learn?

**Comment by mirrormonark** [comment]
yes

**Comment by nischiemineo** [comment]
can you tell me which kind of image do you have? png, jpg, tiff.... and maybe which resolution they have?

**Comment by nischi** [comment]
can you tell me which kind of image do you have? png, jpg, tiff.... and maybe which resolution they have?

**Comment by mirrormonark** [comment]
i took all the photos with the pi-camera using "raspistill -o name.jpg" and saved them to a dataset subfolder with the person's name.after i changed the name file jpg with "img01" at "img15" how say the readme. the saved photos are in jpg format.

**Comment by nischi** [comment]
do you have the filename with a file extension `.jpg`? or only `img01`?

**Comment by mirrormonark** [comment]
i open the folder "dataset" then i open the folder "Skywalker" after i see the file name img01.jpg and its extension is jpg.

**Comment by nischi** [comment]
hmm i will try it again on my pi

**Comment by mirrormonark** [comment]
my raspberry : 
- pi4 8gb
- sistem debian bulleyes 32 bit
- node 20.9 (first it was at 18 then I don't know how it changed to 20)
- npm 10.1
- MagicMirror last version 2.25

**Comment by TroyesDev** [comment]
Same as @mirrormonark , I took some photos in .jpg format, created a dataset folder, a subfolder and named the pictures "img01.jpg" and so on... and am facing the exact same error. Will post info on my system tonight if it might help. Thanks for taking some of your free time to help !

**Comment by nischi** [comment]
Sorry for the late answer. I broke my Pi 🙈

Now i tried it with my new pi and it works just fine. I found an issue on opencv https://github.com/pyinstaller/pyinstaller/issues/6964 with this error. It seems a little bit that the install is corrupt. 
Seems this happen in some versions of opencv

Do you know which version of open cv do you have installed?

**Comment by TroyesDev** [comment]
Thanks you for coming back to us mate ! I was using latest release of opencv (4.8 something), downgrading to latest 4.7 release permitted me to run encode. However when rebooting after adding on config.js i had my classes working but no recognition at all so I tried manually running the recognition in a terminal and ended up having the following message "error on cv2 Waitkey(1) the fonction is not implemented". I tried upgrading to 4.8 again and am facing the previous error once again but this time on recognition.py . Seems like last version of opencv is the problem but I can't run recognition on previous one either. Once again thanks for taking your time to help us !
